### PR TITLE
Add release process section to DEVELOPMENT.md

### PR DESCRIPTION
**Pull request by colebow**
## Description

This PR formalizes the release process into the Trino DEVELOPMENT.md file.

## Non-technical explanation

Updating DEVELOPMENT.md.

## Release notes

(x) This is not user-visible and no release notes are required.
( ) Release notes are required, please propose a release note for me.
( ) Release notes are required, with the following suggested text:

### CODE REVIEW

**Review by mosabua — CHANGES_REQUESTED** [review]
At a minimum we should add links to example issue and PR for a release notes and do some more handwaving around the details of the rest of release build. 

And link to https://github.com/trinodb/release-scripts

File: docs/README.md [hunk 1]
```diff
@@ -215,3 +215,14 @@ contribution](https://trino.io/development/process.html).
...
+
+Trino aims to release once per week on Wednesday. This is a goal but not a
+guarantee, as critical bugs may lead to a release being pushed back or require
+an extra emergency release as a patch. Subsequent releases may also be pushed
+back to the following Wednesday in order to normalize the release cadence.
+
+At the start of each release cycle, a GitHub issue is filed and pinned to track
+all necessary release notes. In addition, a release notes pull request is
+updated and maintained throughout the week, tracking all merged commits to
+ensure every change is properly documented and noted.
```
Comment (findepi) [inline docs/README.md:228]: Belongs to DEVELOPMENT much more than readme

File: docs/README.md [hunk 2]
```diff
@@ -215,3 +215,14 @@ contribution](https://trino.io/development/process.html).
    Docker](https://www.youtube.com/watch?v=y58sb9bW2mA) gives you a starting
    point for setting up a test system on your laptop.
 
+## Release process
+
+Trino aims to release once per week on Wednesday. This is a goal but not a
```
Comment (findepi) [inline docs/README.md:220]: I don't think we want to commit to anything about Wednesdays.

in the general README it should be fine to say that "Trino aims for frequent releases".

File: docs/README.md [hunk 3]
```diff
@@ -215,3 +215,14 @@ contribution](https://trino.io/development/process.html).
    Docker](https://www.youtube.com/watch?v=y58sb9bW2mA) gives you a starting
    point for setting up a test system on your laptop.
 
+## Release process
```
Comment (findepi) [inline docs/README.md:218]: "Release process" usually describes how to do a release (like: run this, publish that, etc.)
This is "Release cadence" or "Releases"

File: .github/DEVELOPMENT.md [hunk 4]
```diff
@@ -170,41 +178,85 @@ Enable errorprone ([Error Prone Installation#IDEA](https://errorprone.info/docs/
...
+documented and noted. This uses the [release note template](../docs/release-template.md),
+with changes in each section arranged to have new features first, performance
+improvements second, and bugfixes third. See [the release notes for 395](https://github.com/trinodb/trino/pull/13975)
+as an example.
+
+Once it is time to release, the release process is kicked off. A code freeze is
+announced on the Trino Slack in the #releases channel, and then the
+[release scripts](https://github.com/trinodb/release-scripts) and the Maven
+release plugin are utilized to update all necessary files and update Trino to
+the next version.
```
Comment (colebow) [inline .github/DEVELOPMENT.md:262]: This is basically word vomit and very hand-wavey. @mosabua let me know if you can go into more detail here, I actually have no clue how a release happens 😆

File: .github/DEVELOPMENT.md [hunk 5]
```diff
@@ -239,3 +239,24 @@ re-compiles when changes to source files are detected:
 To iterate quickly, simply re-build the project in IntelliJ after packaging is
 complete. Project resources will be hot-reloaded and changes are reflected on
 browser refresh.
+
+## Releases
+
+Trino aims for frequent releases, generally once per week. This is a goal but
+not a guarantee, as critical bugs may lead to a release being pushed back or
+require an extra emergency release as a patch.
```
Comment (mosabua) [inline .github/DEVELOPMENT.md:247]: its never a patch .. its a new release

File: .github/DEVELOPMENT.md [hunk 6]
```diff
@@ -239,3 +239,24 @@ re-compiles when changes to source files are detected:
...
+
+At the start of each release cycle, a GitHub issue is filed and pinned to track
+all necessary release notes. For example, see [the issue for Trino 395](https://github.com/trinodb/trino/issues/13913).
+In addition, a release notes pull request is updated and maintained throughout
+the week, tracking all merged commits to ensure every change is properly
+documented and noted. This uses the [release note template](../docs/release-template.md),
+with changes in each section arranged to have new features first, performance
```
Comment (mosabua) [inline .github/DEVELOPMENT.md:254]: Should this be split into a list?

File: .github/DEVELOPMENT.md (hunk 4 above)
Comment (mosabua) [reply .github/DEVELOPMENT.md:262]: I think this is good enough .. could even be cut down to "and then a maintainer performs the  release"

**Review by mosabua — CHANGES_REQUESTED** [review]
Should we add that changes that affect multiple connectors or whatever .. are repeated so they are visible in all relevant sections directly.

Also if a PR fixes multiple things.. there can be multiple release note entries..

File: .github/DEVELOPMENT.md (hunk 6 above)
Comment (colebow) [reply .github/DEVELOPMENT.md:254]: I don't think so, I can't think of a way to make it a reasonable list that isn't super short or jarring.

**Review by mosabua — APPROVED** [review]
I think this is a good small and high level addition.

**Comment by findepi** [comment]
> Add release process section to README 

Worth changing the PR title?

**Comment by colebow** [comment]
@findepi is this good to go? PR title has been updated.

**Review by electrum — APPROVED** [review]

**Pull request merged**